terraform/shared/shared.tf.json.template.py

```diff
@@ -378,9 +382,43 @@ def paren(s: str) -> str:
                 }
             }
         },
-        **(
-            {
-                'aws_cloudformation_stack': {
+        'aws_cloudformation_stack': {
+            # Images whose short name begins with '_' are only used outside the
+            # security boundary, so vulnerabilities that are detected within
+            # them do not need to be addressed with the same urgency.
+            #
+            # Using CF stack because the AWS provider does not support
+            # Inspector filters and the AWSCC provider fails due to
+            # https://github.com/hashicorp/terraform-provider-awscc/issues/1364
+            'inspector_filters': {
+                'name': config.qualified_resource_name('inspectorfilters'),
+                'template_body': json.dumps({
+                    'AWSTemplateFormatVersion': '2010-09-09',
+                    'Description': 'Create suppression rules for select Docker images in AWS Inspector',
+                    'Resources': {
+                        # Must be alphanumeric
+                        hashlib.md5(short_name.encode()).hexdigest(): {
```

Comment:
Why reinvent the wheel here? The precedent I pointed you at in my last review, encapsulates hash-based TF naming functionality in methods of the `ImageRef` class. This resource suppresses findings for an ECR *repository* so the obvious choice is to use the `tf_repository` method of said class.

Reply:
https://github.com/DataBiosphere/azul/pull/5758#discussion_r1427449767

Reply:
Decided at PL to add a new method/property to the `ImageRef` class, and use `sha` instead of `md5`.